New model w cablemount

J-Schaefer opened the pull request:
Added new model to update it to the current real robot in the IAI lab. The model also has the cable guide now that is made of item profiles. Also added some new launch files for the sim and the real robot.

J-Schaefer committed:
Add new unmodified models and launch files

Derived them from the original files and just renamed.

J-Schaefer committed:
Add meshes of item profiles

J-Schaefer committed:
Add new meshes

Added collision models and exchanged the dae files for working
ones. The old dae files were not displayed in Rviz. Solution was
to import them in Blender and export them again.

J-Schaefer committed:
Add item profiles to XML URDF model

The mounting pose is not right yet. Also the inertia property is
missing the right values.

J-Schaefer committed:
Fix poses of the item profiles

The positioning is correct now. Inertia still missing.

J-Schaefer committed:
Add modified meshes

The meshes were improved by merging redundant
vertices using Blender.

J-Schaefer committed:
Add new inertia values to item profile links

The inertia tensor was calculated using the tool MeshLabInertiaToURDF.

J-Schaefer committed:
Add note

J-Schaefer committed:
Edit README and add new launch file for real robot

The new launch file uses the new model with cable guide.

J-Schaefer commented:
I  tested the new launch file with the real robot instead of `/etc/ros/indigo/robot.launch`. I tried moving the robot with the interactive markers and also ran the demo. Collision avoidance also seems to work fine.

ichumuh commented:
giskard seems to be happy with it though

J-Schaefer commented on iai_pr2_description/robots/pr2_with_odom_joints_cableguide.xacro:
If nobody does, then yes. I thought it was safer though. But yeah, no need to keep that much deprecated stuff.

J-Schaefer replied:
The odom version will for now not be deleted since the simulations in iai_pr2_sim are dependent on it. We can delete it in the future since probably nobody uses this simulation anymore.

J-Schaefer commented on iai_pr2_description/robots/pr2_calibrated_with_ft2_cableguide.xml:
1. Was just easier to integrate (this was also how the FT sensor was added). But true. Will change. 
2. URDF is just the file ending right? Or is there a technical difference? But yes, will change when doing a smaller model.
3. Does this matter? It's very simple meshes. Or are boxer still less computational intense?

ichumuh replied:
2. it's fine to combine it, but then we should include it in the old file, or remove the old file. I just don't want multiple urdfs. The ft and ft2 versions also suck. I vote for deleting the ft version to avoid confusion.
3. it doesn't make a huge difference for giskard, but boxes are more efficient than simple meshes. There is no point in adding extra stl files for collision avoidance that is worse.

J-Schaefer replied:
Added new model now that depends on the original `pr2_calibrated_with_ft2.xml`. It's an `*.xacro` like the other models, though. I also removed the stls and added simple boxes.

J-Schaefer commented on iai_pr2_description/launch/upload_pr2_cableguide.launch:
What do you mean? The launch file does include the new URDF or XML in this case. Or do you mean after changing it?

J-Schaefer replied:
What *display.launch* do you mean? Is this resolved?

ichumuh replied:
I meant `roslaunch iai_pr2_description display.launch`

J-Schaefer committed:
Remake model with cable guide as xacro

The model *pr2_with_ft2_cableguide.xacro* now references *pr2_calibrated_with_ft2.xml*
instead of being a full duplicate.

J-Schaefer committed:
Remove collision meshes and replace with boxes

Replaced the collision meshes of the cable guide model with simple boxes.

J-Schaefer committed:
Change command in launch files

Changed the *xacro.py* into *xacro* because it's deprecated since Jade
and *xacro* also works in Indigo.

J-Schaefer committed:
Add real collision mesh for Kinect 1

Added a simple box around the Kinect 1 to prevent self collisions
to the original model *pr2_calibrated_with_ft2.xml*.

J-Schaefer commented:
I noticed that the PR2 is still crashing with the camera into the pole. I already added a new collision mesh for the Kinect 1, since it didn't have one. Also need to add a collision (and visual) mesh for the Kinect 2.

J-Schaefer committed:
Add Kinect2 to the model

J-Schaefer commented:
Added Kinect 2 to the model.

ichumuh commented:
I can also just push it if you want.
Otherwise giskard is happy with the new urdf.
